src/main/java/com/wisedevlife/whytalkuser/dto/response/UserProfileResponse.java

```diff
@@ -2,12 +2,28 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.wisedevlife.whytalkuser.common.enums.SexEnum;
-import java.util.Date;
+import com.wisedevlife.whytalkuser.common.utils.UserProfileUtil;
+import com.wisedevlife.whytalkuser.entity.UserProfile;
+import java.time.LocalDate;
+import lombok.Builder;
 import lombok.Data;
 import org.springframework.format.annotation.DateTimeFormat;
 
 @Data
+@Builder
 public class UserProfileResponse {
+    public static UserProfileResponse of(UserProfile userProfile) {
+        UserProfileUtil profileUtil = UserProfileUtil.builder().build();
+        Integer userAge = profileUtil.convertBirthDayToAge(userProfile.getBirthDay());
+        return UserProfileResponse.builder()
+                .userId(userProfile.getUserId())
+                .sex(userProfile.getSex())
+                .displayName(userProfile.getDisplayName())
+                .birthDay(userProfile.getBirthDay())
+                .age(userAge)
+                .build();
+    }
+
     @JsonProperty("user_id")
```

Comment:
Maybe we could use the `@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class`) annotation in the request and response payloads for our JSON naming convention.